tests/unit_tests/test_parquet.py

```diff
@@ -0,0 +1,93 @@
+import sys
+from contextlib import nullcontext
+from pathlib import Path
+
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+import pytest
+import scipy
+
+if "typeguard" in sys.modules:
+    from typeguard import suppress_type_checks
+
+from spectrum_io.file import parquet
+
+
+class TestParquet:
+    """Test class to check Parquet file I/O."""
+
+    def test_read_file(self, raw_data, tmpdir):
+        """Test read operation for a single dataset."""
+        output_path = Path(tmpdir / "table.parquet")
+        pq.write_table(pa.Table.from_pydict(raw_data), output_path)
+        df = parquet.read_file(output_path)
+        pd.testing.assert_frame_equal(df, pd.DataFrame(raw_data))
+
+    def test_write_file(self, raw_data, tmpdir):
+        """Check write operation for a single dataset."""
+        output_path = Path(tmpdir / "table.parquet")
+        df = pd.DataFrame(raw_data)
+        parquet.write_file(df, output_path)
+        pd.testing.assert_frame_equal(df, pd.read_parquet(output_path))
+
+    def test_read_write_partition(self, raw_data, tmpdir):
+        """Check whether data is unmodified after being written to and then read from a partitioned dataset."""
+        output_path = Path(tmpdir / "partition")
+        df = pd.DataFrame(raw_data)
+        parquet.write_partition([df, df], output_path, ["dataset_1", "dataset_2"])
+        read_df = parquet.read_partition(output_path, "dataset_1")
+        pd.testing.assert_frame_equal(read_df, df)
+
+    def test_read_write_partition_integer_key(self, raw_data, tmpdir):
+        """Check whether Parquet's under-the-hood conversion of string to integer keys is handled seamlessly."""
+        output_path = Path(tmpdir / "partition")
+        df = pd.DataFrame(raw_data)
+        parquet.write_partition([df, df], output_path, ["1", "2"])
+        read_df = parquet.read_partition(output_path, "1")
+        pd.testing.assert_frame_equal(read_df, df)
+
+    def test_modify_partition(self, raw_data, tmpdir):
+        """Check whether file content stays the same when writing new data to the same partitioned directory."""
+        output_path = Path(tmpdir / "partition")
+        df = pd.DataFrame(raw_data)
+        parquet.write_partition([df, df], output_path, ["1", "2"])
+        parquet.write_partition([df, df, df], output_path, ["1", "2", "3"])
+        read_df = parquet.read_partition(output_path, "2")
+        pd.testing.assert_frame_equal(read_df, df)
+
+    def test_write_not_implemented(self, raw_data, tmpdir):
+        """Check whether write_file() raises a NotImplementedError if provided with an unsupported object."""
+        with pytest.raises(NotImplementedError):
```

Comment:
Can you elaborate on this a bit? How do you want to test for raising on top of `pytest.raises`? Write a custom `assert_raises` context manager?

Reply:
nevermind, finally found `unittest`'s `assertRaises` after some googling around. Would there be any advantage of inheriting `unittest.TestCase` and using `assertRaises` instead of `pytest.raises()`? I can see that some test files (e.g. `test_xisearch.py` have classes that inherit from it but don't seem to use any specific features, whereas others like `TestMSP` in `test_spectral_library.py` do not

Reply:
ok, we leave it as is.